trench/command/deactivate_mfa_method.py

```diff
@@ -14,7 +14,8 @@ def __init__(self, mfa_model: Type[MFAMethod]) -> None:
     @atomic
     def execute(self, mfa_method_name: str, user_id: int) -> None:
         mfa = self._mfa_model.objects.get_by_name(user_id=user_id, name=mfa_method_name)
-        if mfa.is_primary:
+        active_mfa_methods = self._mfa_model.objects.list_active(user_id=user_id)
+        if mfa.is_primary and len(active_mfa_methods) > 1:
```

Comment:
It's much more efficient to return a scalar from the database than to return rows, map them to Python objects just to check the length of the objects list at the end 😉 

```suggestion
        number_of_active_mfa_methods = self._mfa_model.objects.filter(user_id=user_id, is_active=True).count()
        if mfa.is_primary and number_of_active_mfa_methods > 1:
```